lib/query_builder/schema/comparison_filter.ex

```diff
@@ -34,6 +34,18 @@ defmodule EctoShorts.QueryBuilder.Schema.ComparisonFilter do
     where(query, [scm], field(scm, ^filter_field) == ^val)
   end
 
+  def build_array(query, filter_field, val) when is_list(val) do
+    where(query, [scm], field(scm, ^filter_field) == ^val)
+  end
+
+  def build_array(query, filter_field, filters) when is_map(filters) do
+    build(query, filter_field, filters)
```

Comment:
What's the behaviour here? If it's a map what happens?

Reply:
This is to ensure standard filter behaviour. If a map is passed we send the call to build where the is_map guard will pick it up and do the standard build_subfield_filter reduce. 

This is to ensure we can still make calls to check if the field is {:==, nil}, {:!=, nil}, {:!=, val}

Reply:
How does the current code handle this? And how would it need to change since the code that previously did this will no longer be doing this from what I gather? Should we change the method name to reflect that it's not just buliding and array?

Reply:
So this is preserving the behaviour of the current code
`CommonFilters.convert_params_to_filter(User, %{array_field: %{!=: nil}})`
in the current code this would go to the build function and get picked up by the is_map guard

Reply:
Okay cool! Looks good!